frontend/src/store/auth/index.js

```diff
@@ -1,107 +1,81 @@
-    })
-}
+import {createUser} from '@/utils/api/auth'
 
 const state = {
   loading: false,
-  error: '',
-  user: null
+  error: ''
 }
 
 const getters = {
   loading: ({ loading }) => loading,
-  isLoggedIn: ({ user }) => !!user,
-  error: ({ error }) => error,
-  user: ({ user }) => user
+  error: ({ error }) => error
 }
 
 const mutations = {
   ERROR(state, message) {
-    console.log('ERROR', message)
+    if (message) console.log('AUTH ERROR', message)
     state.error = message
   },
-  SIGN_UP(state, user) {
-    // on successful signup communicate with backend to update database.
-    // If by email, new users are signed in by default
-    console.log('SIGN UP', user.user)
-    sendAuthToken(user.user)
-    state.user = user.user
-  },
-  LOGIN(state, user) {
-    // login happens in indexedDB in chrome, localStorage in other browsers
-    console.log('LOGIN', user.user)
-    sendAuthToken(user.user)
-    state.user = user.user
-  },
-  LOGOUT(state) {
-    state.user = null
-    console.log('LOGOUT EVENT')
-  },
-  UPDATE(state, user) {
-    // useful for intial boot
-    console.log(user)
-    if (user) sendAuthToken(user)
-    state.user = user
+  SET_LOADING(state, isLoading) {
+    state.loading = isLoading
   }
 }
 
+/* successful signIn returns an UserAuth object which has field user */
 const actions = {
   signIn({commit}, { email, password }) {
-    if (!(email && password)) {
-      return commit('ERROR', 'Missing fields')
-    }
-
+    commit('SET_LOADING', true)
     return auth.signInWithEmailAndPassword(email, password)
-      .then(user => commit('LOGIN', user))
+      .then(({ user }) => commit('SET_USER', user, {root: true}))
       .catch(error => commit('ERROR', error.message))
+      .finally(() => commit('SET_LOADING', false))
   },
 
+  /**
+   * logout client side. Everything will be handled by
+   * firebase for this.
+   */
   logout({commit}) {
     return auth.signOut()
-      .then(() => commit('LOGOUT'))
+      .then(() => commit('SET_USER', null, {root: true}))
       .catch(error => commit('ERROR', error.message))
   },
 
+  /**
+   * successful signup returns an UserAuth object.
+   * UserAuth obj has field user which is what we're interested in
+   * Currently, successful signUp will automatically sign in user.
+   **/
   signUp({commit}, { email, password }) {
-    if (!(email && password)) {
-      return commit('ERROR', 'Missing fields')
-    }
-
+    commit('SET_LOADING', true)
     return auth.createUserWithEmailAndPassword(email, password)
-      .then(user => commit('SIGN_UP', user))
+      .then(({ user }) => {
+        /* TODO implement retry if failure occurs on server
+          Simultaneously:
+          1. Send off new user creds for backend creation,
+          2. Set user state in store.
+         */
+        return Promise.all([
+          createUser(user),
+          commit('SET_USER', user, {root: true})])
+      })
       .catch(error => commit('ERROR', error.message))
```

Comment:
Probs need a TODO here to be able to display an error back to the user? e.g. atm if you provide a password less than 6 characters you only know based on the console message.

Reply:
Yes?